### PR TITLE
[DEV-4229] Recursively load team members in analytics

### PR DESCRIPTION
**Pull request by gaetano-guerriero**

### CODE REVIEW

File: server/athenian/api/controllers/team_controller.py [hunk 1]
```diff
@@ -299,6 +300,63 @@ async def get_team_from_db(
     return team
 
 
+async def teams_recursive_fetch(
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/team_controller.py:309]:
```suggestion
async def fetch_teams_recursively(
```

File: server/athenian/api/controllers/team_controller.py [hunk 2]
```diff
@@ -305,6 +306,63 @@ async def get_team_from_db(
...
+    account_id: int,
+    sdb_conn: DatabaseLike,
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/team_controller.py:311]:
```suggestion
    sdb: DatabaseLike,
```

File: server/athenian/api/controllers/team_controller.py [hunk 3]
```diff
@@ -305,6 +306,63 @@ async def get_team_from_db(
     return team
 
 
+async def teams_recursive_fetch(
+    account_id: int,
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/team_controller.py:310]:
```suggestion
    account: int,
```
(common naming in the project)

File: server/athenian/api/controllers/team_controller.py [hunk 4]
```diff
@@ -305,6 +306,63 @@ async def get_team_from_db(
...
+
+    If `root_team_ids` is passed those will be taken as base for the recursion.
+    If `root_team_ids` is None all root teams will be taken as base for the recursion.
+
+    The returned list of teams will include duplicates when
+    on of the `root_team_ids` is ancestor of another.
+
+    Returned columns can be selected with `select_entities`. The ID of the root team
+    used to fetch the row will always be included as last column.
+
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/team_controller.py:326]:
```suggestion
```

File: server/athenian/api/controllers/team_controller.py [hunk 5]
```diff
@@ -305,6 +306,63 @@ async def get_team_from_db(
...
+    # Team.id is required inside cte for join even if not requested by caller
+    if Team.id not in select_entities:
+        cte_select_entities: Sequence[InstrumentedAttribute] = (*select_entities, Team.id)
+    else:
+        cte_select_entities = select_entities
+
+    # base team is the specified root_team_id if present, else all account's root teams
+    recursive_base_where = Team.owner_id == account_id
+    if root_team_ids is None:
+        recursive_base_where = sa.and_(recursive_base_where, Team.parent_id == None)  # noqa: E711
```
Comment (vmarkovtsev) [inline server/athenian/api/controllers/team_controller.py:339]:
```suggestion
        recursive_base_where = sa.and_(recursive_base_where, Team.parent_id.is_(None))
```

Comment (gaetano-guerriero) [reply server/athenian/api/controllers/team_controller.py]: nice, it was horrible to add the flake8 override

**Review by vmarkovtsev — APPROVED** [review]

**Pull request merged**